### PR TITLE
Adding zip and unzip operations

### PR DESCRIPTION
**Pull request by raskin-aleksandr**
Signed-off-by: raskin-aleksandr <raskin.aleksandr@gmail.com>

### CODE REVIEW

File: org/openscore/slang/base/files/unzip_archive.sl [hunk 1]
```diff
@@ -0,0 +1,23 @@
+namespace: task.zip
```
Comment (meirwah) [inline org/openscore/slang/base/files/unzip_archive.sl]: documentation is needed...

File: org/openscore/slang/base/files/zip_folder.sl [hunk 2]
```diff
@@ -0,0 +1,29 @@
+namespace: task.zip
```
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: documentation is needed...

File: org/openscore/slang/base/files/unzip_archive.sl (hunk 1 above)
Comment (meirwah) [inline org/openscore/slang/base/files/unzip_archive.sl]: namespace should be the same as folder structure..

File: org/openscore/slang/base/files/zip_folder.sl (hunk 2 above)
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: namespace should be the same as folder structure..

File: org/openscore/slang/base/files/zip_folder.sl [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          filename = archive_name + '.zip'
+          if os.path.isdir(output_folder):
+            shutil.move(filename, output_folder)
+          else:
+            print ("else")
+            os.mkdir(output_folder)
+            shutil.move(filename, output_folder)
+          res = 'True'
+        except Exception:
+          print sys.exc_info()[0]
```
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: no need to print error, instead add an output of the error

File: org/openscore/slang/base/files/zip_folder.sl [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          res = 'False'
```
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: i would change "res" to a more meaningful name..

File: org/openscore/slang/base/files/unzip_archive.sl [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    - archive_name
+    - output_folder
+
+  action:
+    python_script: |
+        import zipfile, sys
+        try:
+          with zipfile.ZipFile(archive_name, "r") as z:
+            z.extractall(output_folder)
+          res = 'True'
```
Comment (orius123) [inline org/openscore/slang/base/files/unzip_archive.sl]: You can use actual booleans here

File: org/openscore/slang/base/files/zip_folder.sl [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          filename = archive_name + '.zip'
+          if os.path.isdir(output_folder):
+            shutil.move(filename, output_folder)
+          else:
+            os.mkdir(output_folder)
+            shutil.move(filename, output_folder)
+          result = True
+        except Exception:
+          error_messsage = sys.exc_info()[0]
+          reselut = False
```
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: typo here : reselut
please test negative case...

File: org/openscore/slang/base/files/zip_folder.sl [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+  outputs:
+    - error_messsage: "'Something went wrong: ' + error_messsage"
```
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: I would skip the 'Something went wrong: ' prefix...
and what happens when error_messsage is empty? does the flow works?
maybe init the var error_messsage='' , atthe start of the script...

File: org/openscore/slang/base/files/zip_folder.sl [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+            os.mkdir(output_folder)
+            shutil.move(filename, output_folder)
+          message = "'zip created successfully'"
+          result = True
+        except Exception:
+          messsage = sys.exc_info()[0]
+          result = False
+
+  outputs:
+    - messsage: messsage
```
Comment (meirwah) [inline org/openscore/slang/base/files/zip_folder.sl]: typo : messsage

**Pull request merged**